### PR TITLE
Remember me function, Exception handling for connection timeouts,Error pages

### PR DESCRIPTION
**Pull request by Yasara123**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/org.wso2.carbon.status.dashboard.web/src/auth/Login.jsx [hunk 1]
```diff
@@ -62,7 +62,7 @@ export default class Login extends Component {
         this.authenticate = this.authenticate.bind(this);
     }
 
-    componentWillMount() {
+    componentWillMount(){
```
Comment (gimantha) [inline components/org.wso2.carbon.status.dashboard.web/src/auth/Login.jsx]: add space before {

File: components/org.wso2.carbon.status.dashboard.web/src/auth/utils/AuthManager.jsx [hunk 2]
```diff
@@ -210,31 +187,11 @@ export default class AuthManager {
...
-    }
-
     /**
      * Delete a browser cookie given its name
      * @param {String} name : Name of the cookie which need to be deleted
      */
-    static delete_cookie(name) {
-        document.cookie = name + '=; Path=' + window.contextPath + '; Expires=Thu, 01 Jan 1970 00:00:01 GMT;';
+    static deleteCookie(name) {
+        document.cookie = name + '=; path=' + window.contextPath + '; expires=Thu, 01 Jan 1970 00:00:01 GMT;';
```
Comment (gimantha) [inline components/org.wso2.carbon.status.dashboard.web/src/auth/utils/AuthManager.jsx]: formatting issues - proper whitespaces before and after the operator

File: components/org.wso2.carbon.status.dashboard.web/src/overview/WorkerOverview.jsx [hunk 3]
```diff
@@ -88,25 +102,29 @@ export default class WorkerOverview extends React.Component {
...
-            let found = error.response.data.match(re);
-            if (found != null) {
-                this.setState({
-                    isApiCalled: true,
-                    sessionInvalid: true,
-                    statusMessage: "User session Invalid, Please Login"
-                })
-            } else {
+            let message;
+            if(error.response != null){
```
Comment (gimantha) [inline components/org.wso2.carbon.status.dashboard.web/src/overview/WorkerOverview.jsx]: space before {

Comment (gimantha) [reply components/org.wso2.carbon.status.dashboard.web/src/overview/WorkerOverview.jsx]: check other places also

File: components/org.wso2.carbon.status.dashboard.web/src/siddhi-app/AppSpecific.jsx [hunk 4]
```diff
@@ -164,34 +164,58 @@ export default class WorkerSpecific extends React.Component {
...
             .then((response) => {
                 that.setState({
                     hasManagerPermission: response.data
                 });
             }).catch((error) => {
-            let re = /The session with id '((?:\\.|[^'])*)'|"((?:\\.|[^"])*)" is not valid./;
-            let found = error.response.data.match(re);
-            if (found != null) {
+            let message;
+            if(error.response != null){
```
Comment (gimantha) [inline components/org.wso2.carbon.status.dashboard.web/src/siddhi-app/AppSpecific.jsx]: space before {.. check other places in this file

File: components/org.wso2.carbon.status.dashboard.web/src/siddhi-app/ComponentHistory.jsx [hunk 5]
```diff
@@ -352,43 +389,47 @@ export default class ComponentHistory extends React.Component {
...
-                        to={window.contextPath + '/worker/' + this.props.match.params.id + '/siddhi-apps/' +
-                        this.props.match.params.appName + "/" + this.state.statsEnable}>
-                        <FlatButton label={this.props.match.params.appName + " >"}/>
-                    </Link>
-                    <RaisedButton label={this.props.match.params.componentId} disabled disabledLabelColor='white'
-                                  disabledBackgroundColor='#f17b31'/>
-                </div>
-                <div className="worker-h1">
-                    <h2 style={{marginLeft: 40}}> {this.props.match.params.componentId} Metrics </h2>
+        if(this.state.hasViewerPermission) {
```
Comment (gimantha) [inline components/org.wso2.carbon.status.dashboard.web/src/siddhi-app/ComponentHistory.jsx]: Check whther proper whitespaces are used.. (formatting issues)

File: components/org.wso2.carbon.status.dashboard.web/src/worker/WorkerSpecific.jsx [hunk 6]
```diff
@@ -68,26 +69,58 @@ export default class WorkerSpecific extends React.Component {
...
             .then((response) => {
                 that.setState({
                     hasManagerPermission: response.data
                 });
             }).catch((error) => {
-            let re = /The session with id '((?:\\.|[^'])*)'|"((?:\\.|[^"])*)" is not valid./;
-            let found = error.response.data.match(re);
-            if (found != null) {
+            let message;
+            if(error.response != null){
```
Comment (gimantha) [inline components/org.wso2.carbon.status.dashboard.web/src/worker/WorkerSpecific.jsx]: proper spacings are missing

Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.web/src/worker/WorkerSpecific.jsx]: Fixed in 50c0b3ff27410966326b69c3e721733fe60b591b

File: components/org.wso2.carbon.status.dashboard.web/src/siddhi-app/ComponentHistory.jsx (hunk 5 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.web/src/siddhi-app/ComponentHistory.jsx]: Fixed in 50c0b3ff27410966326b69c3e721733fe60b591b

File: components/org.wso2.carbon.status.dashboard.web/src/siddhi-app/AppSpecific.jsx (hunk 4 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.web/src/siddhi-app/AppSpecific.jsx]: Fixed in b4a2bcf444d25031775a679bad76798d70ce7d5b

File: components/org.wso2.carbon.status.dashboard.web/src/overview/WorkerOverview.jsx (hunk 3 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.web/src/overview/WorkerOverview.jsx]: Fixed in b4a2bcf444d25031775a679bad76798d70ce7d5b

**Pull request merged**